js/components/SocialShare/index.js

```diff
@@ -0,0 +1,58 @@
+/* global require */
+import React, { Component, PropTypes } from 'react';
+import {
+  TouchableOpacity,
+  Image,
+  StyleSheet,
+} from 'react-native';
+import Share from 'react-native-share';
+
+export default class SocialShare extends Component {
+  constructor(props) {
+    super(props);
+
+    this.share = this.share.bind(this);
+  }
+
+  share() {
+    let shareOptions = this.props.itemId ? 
+      this.fromStore() : 
+      Object.assign({}, this.props);
```

Comment:
It isn't clear what use-cases your supporting here. It looks as though the `itemId` is optional and impacts the properties that are passed to the share intent. But I'm not sure why?

I'd prefer we just be super explicit (no internal magic) and require that props are always passed into the component to it to work.

Right now we only have the 'share item' use-case specced so let's not add any code that's not required.